torchtext/datasets/text_classification.py

```diff
@@ -0,0 +1,416 @@
+            root: Directory where the dataset are saved. Default: ".data"
+            text_field: The field that will be used for the sentence. If not given,
+                'spacy' token will be used.
+            label_field: The field that will be used for the label. If not given,
+                'float' token will be used.
+            ngrams: a contiguous sequence of n items from s string text.
+                Default: 1
+
+        Examples:
+            >>> url = 'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbUDNpeUdjb0wxRms'
+            >>> txt_cls = TextClassificationDataset(url, ngrams=2)
+
+        """
+        fields = []
+        fields.append(('text', text_field if text_field is not None
+                       else data.Field(tokenize=data.get_tokenizer('spacy'),
+                                       init_token='<SOS>',
+                                       eos_token='<EOS>')))
+        fields.append(('label', label_field if label_field is not None
+                       else data.LabelField(dtype=torch.float)))
+
+        super(TextClassificationDataset, self).__init__([], dict(fields))
+
+        self.dataset_name = self.__class__.__name__
+        self.root = root
+        self.raw_folder = os.path.join(self.root, self.__class__.__name__, 'raw')
+        self.processed_folder = os.path.join(self.root,
+                                             self.__class__.__name__,
+                                             'processed')
+        filepath = os.path.join(self.processed_folder, self.dataset_name + '.train')
+        if not os.path.isfile(filepath):
+            download(self.url, self.raw_folder, self.dataset_name)
+            preprocess(self.raw_folder, self.processed_folder, self.dataset_name)
+        self.train_examples = load_text_classification_data(filepath, self.fields, ngrams)
+        filepath = os.path.join(self.processed_folder, self.dataset_name + '.test')
+        self.test_examples = load_text_classification_data(filepath, self.fields, ngrams)
+        self.examples = self.train_examples + self.test_examples
+        self.fields['text'].vocab = build_vocab(self, self.fields['text'], 'text')
+        self.fields['label'].vocab = build_vocab(self, self.fields['label'], 'label')
+
+    @staticmethod
+    def sort_key(ex):
+        return len(ex.text)
+
+
+class AG_NEWS(TextClassificationDataset):
+    """ Defines AG_NEWS datasets.
+        The labels includes:
+            - 1 : World
+            - 2 : Sports
+            - 3 : Business
+            - 4 : Sci/Tech
+     """
+    def __init__(self, root='.data', text_field=None, label_field=None, ngrams=1):
+        """Create supervised learning dataset: AG_NEWS
+
+        Inputs:
+            See TextClassificationDataset() class
+
+        Examples:
+            >>> text_cls = torchtext.datasets.AG_NEWS(ngrams=3)
+
+        """
+        self.url = 'https://drive.google.com/uc?export=download&id=0Bz8a_Dbh9QhbUDNpeUdjb0wxRms'
```

Comment:
Can you expose this as part of a static list? I think this is something that could be useful by itself. The user can then also use it as a reference for other programs via something like torchaudio.datasets.supervised.URLS['AG_NEWS']

Reply:
Sure. I haven't updated the APIs to address your comments above. Will do them all together.

Reply:
Fixed